cpp/src/DataStorm/SessionI.cpp

```diff
@@ -667,6 +667,13 @@ SessionI::retry(NodePrx node, exception_ptr exception)
         {
             rethrow_exception(exception);
         }
+        catch (const DataStormContract::SessionCreationException& ex)
+        {
+            if (ex.error == SessionError::NodeShutdown)
+            {
+                return false;
+            }
```

Comment:
else we ignore this exception. You should add a comment explaining why, to make it clear it's on purpose.